File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # temp=[]
        # for i in range(len(nums)-1):
        #     for j in range(i+1,len(nums)):
        #         if nums[i]+nums[j]==target:
        #             temp.append(i) 
        #             temp.append(j) 
        #         else:
        #             continue
        # return temp
        # n=len(nums)
        # m={}
        # res=[]
        # for i in range(n):
        #     x=target - nums[i]
        #     if x not in m:
        #         m[nums[i]]=i
        #     else:
        #         res.append(m[x])
        #         res.append(i)
        # return res

        m={}
        res=[]
        for i in range(len(nums)):
            x=target-nums[i]
            if x not in m:
                m[nums[i]]=i
            else:
                res.append(m[x])
                res.append(i)
        return res

            
                
        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let m={};
    let t=[];
    for (let i=0;i<nums.length;i++){
        let x=target-nums[i];
        if (!(x in m)){
            m[nums[i]]=i
        }else{
            t.push(m[x]);
            t.push(i);        
            return t;
        }
    }
};
//   let map = new Map()
//   let  n= nums.length

//   for(let i = 0 ; i<n ; i++){

//     let k = target - nums[i];

//     if(map.has(k)){
//         return [map.get(k) ,i];
//     }

//     map.set(nums[i] , i)
//   }

  
    